test/reduce.jl

```diff
@@ -94,6 +94,9 @@ end
 
 @test sum_kbn([1,1e100,1,-1e100]) == 2
 @test sum_kbn(Float64[]) == 0.0
+@test sum_kbn((i for i=1.0:1.0:10.0)) == 55.0
+@test sum_kbn((i for i=1:1:10)) == 55
+@test sum_kbn([1 2 3]) == 6
 
```

Comment:
Also, there's no need for double parentheses when writing generators inside a function.